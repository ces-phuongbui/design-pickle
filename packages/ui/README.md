# UI Component Library

This package contains reusable UI components for the CES Design Pickle project.

## Testing

This package uses Jest and React Testing Library for unit testing components.

### Running Tests

To run the tests, use the following commands:

```bash
# Run tests once
pnpm test

# Run tests in watch mode (useful during development)
pnpm test:watch

# Generate test coverage report
pnpm test:coverage
```

### Test Coverage

The test coverage report will be generated in the `coverage` directory. You can open `coverage/lcov-report/index.html` in your browser to view a detailed report.

The coverage report shows:
- Statement coverage: The percentage of statements that have been executed
- Branch coverage: The percentage of branches (e.g., if/else) that have been executed
- Function coverage: The percentage of functions that have been called
- Line coverage: The percentage of executable lines that have been executed

### Writing Tests

Tests are located in the `src/components/__tests__` directory. Each component should have a corresponding test file with the naming convention `ComponentName.test.tsx`.

For more detailed information about testing, including examples and best practices, see the [TESTING.md](./TESTING.md) file.
