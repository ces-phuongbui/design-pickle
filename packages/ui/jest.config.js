module.exports = {
  preset: 'ts-jest',
  testEnvironment: 'jsdom',
  moduleNameMapper: {
    // Handle CSS imports (with CSS modules)
    '^.+\\.module\\.(css|sass|scss)$': 'identity-obj-proxy',
    // Handle CSS imports (without CSS modules)
    '^.+\\.(css|sass|scss)$': '<rootDir>/__mocks__/styleMock.js',
    // Handle image imports
    '^.+\\.(jpg|jpeg|png|gif|webp|svg)$': '<rootDir>/__mocks__/fileMock.js',
  },
  setupFilesAfterEnv: ['<rootDir>/jest.setup.js'],
  testPathIgnorePatterns: ['<rootDir>/node_modules/'],
  transform: {
    '^.+\\.(ts|tsx)$': ['ts-jest', {
      tsconfig: 'tsconfig.json',
      babelConfig: {
        presets: ['@babel/preset-react']
      }
    }],
  },
  transformIgnorePatterns: [
    '/node_modules/(?!(@radix-ui|lucide-react)/)',
  ],
  coverageDirectory: 'coverage',
  collectCoverageFrom: [
    'src/**/*.{ts,tsx}',
    '!src/**/*.d.ts',
    '!src/**/*.stories.{ts,tsx}',
    '!src/index.ts',
  ],
}; 