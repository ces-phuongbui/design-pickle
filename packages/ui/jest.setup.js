// Add custom jest matchers for testing-library
require("@testing-library/jest-dom");
