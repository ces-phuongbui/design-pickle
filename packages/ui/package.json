{
  "name": "@repo/ui",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "ui": "pnpm dlx shadcn@latest",
    "lint": "eslint .",
    "test": "jest",
    "test:watch": "jest --watch",
    "test:coverage": "jest --coverage"
  },
  "peerDependencies": {
    "next": "^15",
    "react": "^18"
  },
  "devDependencies": {
    "@babel/preset-react": "^7.26.3",
    "@repo/eslint-config": "workspace:*",
    "@repo/typescript-config": "workspace:*",
    "@testing-library/jest-dom": "^6.4.2",
    "@testing-library/react": "^14.2.1",
    "@testing-library/user-event": "^14.5.2",
    "@types/jest": "^29.5.12",
    "@types/node": "catalog:",
    "@types/react": "catalog:react18",
    "autoprefixer": "^10",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^29.7.0",
    "jest-environment-jsdom": "^29.7.0",
    "postcss": "^8",
    "postcss-load-config": "^6",
    "react": "18.3.1",
    "react-dom": "18.3.1",
    "tailwindcss": "catalog:",
    "ts-jest": "^29.1.2",
    "typescript": "catalog:"
  },
  "dependencies": {
    "@radix-ui/react-accordion": "^1.2.3",
    "@radix-ui/react-avatar": "^1.1.3",
    "@radix-ui/react-dialog": "^1.1.6",
    "@radix-ui/react-dropdown-menu": "^2.1.6",
    "@radix-ui/react-icons": "^1.3.2",
    "@radix-ui/react-navigation-menu": "^1.2.5",
    "@radix-ui/react-slot": "^1.0.2",
    "@strapi/blocks-react-renderer": "^1.0.2",
    "@strapi/client": "^1.0.1",
    "class-variance-authority": "^0.7.0",
    "clsx": "^2.1.1",
    "embla-carousel-react": "^8.5.2",
    "lucide-react": "catalog:",
    "tailwind-merge": "^2.3.0",
    "tailwindcss-animate": "^1.0.7"
  },
  "exports": {
    "./globals.css": "./src/globals.css",
    "./postcss.config": "./postcss.config.mjs",
    "./tailwind.config": "./tailwind.config.ts",
    "./assets/*": "./src/assets/*.*",
    "./lib/*": "./src/lib/*.ts",
    "./hooks/*": [
      "./src/hooks/*.ts",
      "./src/hooks/*.tsx"
    ],
    "./types/*": "./src/types/*.ts",
    "./components/*": "./src/components/*.tsx"
  }
}
