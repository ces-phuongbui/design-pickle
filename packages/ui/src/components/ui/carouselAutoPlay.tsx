import { cn } from "@repo/ui/lib/utils";
import { Carousel, CarouselContent } from "./carousel";

export const CarouselAutoPlay = ({
  className,
  children,
}: {
  className?: string;
  children: React.ReactNode;
}) => {
  return (
    <Carousel
      className={cn("w-full", className)}
      opts={{
        loop: true,
      }}
      autoplay={true}
      autoplayInterval={2500}
    >
      <CarouselContent>{children}</CarouselContent>
    </Carousel>
  );
};
