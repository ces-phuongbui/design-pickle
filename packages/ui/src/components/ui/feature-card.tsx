import Link from "next/link";
import { StrapiImage } from "../StrapiImage";
import { Button } from "./button";

export interface FeatureCardProps {
  title: string;
  description: string;
  backgroundImage?: string;
  ctaText?: string;
  ctaHref?: string;
}

export const FeatureCard = ({
  title,
  description,
  backgroundImage,
  ctaText,
  ctaHref,
}: FeatureCardProps) => {
  return (
    <div
      data-testid="feature-card"
      className="relative h-[600px] overflow-hidden rounded-lg bg-gradient-to-b from-purple-600 to-teal-500 p-6 text-white"
    >
      {/* Background Image with Gradient Overlay */}
      <div className="absolute inset-0 z-0">
        {backgroundImage ? (
          <StrapiImage
            alt={title}
            src={backgroundImage}
            width={400}
            height={600}
            className="h-full w-full object-cover"
          />
        ) : (
          <div className="absolute inset-0 bg-gradient-to-b from-transparent via-purple-600/50 to-teal-900/90" />
        )}
      </div>

      {/* Content */}
      <div className="relative z-10 flex h-full flex-col justify-between">
        <h3 className="text-3xl font-bold">{title}</h3>

        <div className="space-y-4">
          <p className="text-xl text-gray-100 mb-5">{description}</p>
          <Link href={ctaHref ?? ""}>
            <Button
              variant="link"
              className="group px-0 text-white hover:no-underline bg-white bg-opacity-20 p-4"
            >
              {ctaText ?? "Read more"}{" "}
              <span className="ml-2 transition-transform group-hover:translate-x-1">
                â†’
              </span>
            </Button>
          </Link>
        </div>
      </div>
    </div>
  );
};
