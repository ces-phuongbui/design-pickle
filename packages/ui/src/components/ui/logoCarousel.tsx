import { cn } from "@repo/ui/lib/utils";
import { StrapiImage } from "../StrapiImage";

interface LogoCarouselProps {
  logoUrl: string;
  name: string;
  className?: string;
}

export const LogoCarousel = ({
  logoUrl,
  name,
  className,
}: LogoCarouselProps) => {
  return (
    <div
      className={cn(
        "flex items-center bg-transparent p-4 rounded-md space-x-3",
        className,
      )}
    >
      <div className="relative w-8 h-8">
        <StrapiImage
          alt={`${name} logo`}
          src={logoUrl}
          width={100}
          height={100}
          className="object-contain"
        />
      </div>
      <span className="text-white text-lg font-medium">{name}</span>
    </div>
  );
};
