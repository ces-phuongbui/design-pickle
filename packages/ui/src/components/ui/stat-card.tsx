import { CardStat } from "@repo/ui/types";

export const StatCard = ({ value, label, subLabel }: CardStat) => {
  return (
    <div
      data-testid="stat-card"
      className="flex flex-col items-center space-y-2 rounded-lg bg-zinc-900 p-6 text-center text-white"
    >
      <div className="text-3xl font-bold md:text-4xl lg:text-5xl">{value}</div>
      <div className="space-y-1">
        {label && (
          <div className="text-base text-gray-200 md:text-lg font-bold">
            {label}
          </div>
        )}
        {subLabel && <div className="text-sm text-gray-400">{subLabel}</div>}
      </div>
    </div>
  );
};
