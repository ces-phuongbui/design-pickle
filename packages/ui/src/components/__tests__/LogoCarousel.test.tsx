import { render, screen } from "@testing-library/react";
import { LogoCarousel } from "../ui/logoCarousel";

describe("LogoCarousel", () => {
  const mockProps = {
    logoUrl: "https://example.com/logo.png",
    name: "Test Logo",
    className: "test-class",
  };

  it("renders correctly and matches snapshot", () => {
    const { container } = render(<LogoCarousel {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it("displays logo name", () => {
    render(<LogoCarousel {...mockProps} />);
    expect(screen.getByText(mockProps.name)).toBeInTheDocument();
  });

  it("applies custom className", () => {
    const { container } = render(<LogoCarousel {...mockProps} />);
    expect(container.firstChild).toHaveClass("test-class");
  });
});
