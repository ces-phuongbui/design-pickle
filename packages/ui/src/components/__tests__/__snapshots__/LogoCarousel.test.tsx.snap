// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoCarousel renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="flex items-center bg-transparent p-4 rounded-md space-x-3 test-class"
  >
    <div
      class="relative w-8 h-8"
    >
      <img
        alt="Test Logo logo"
        class="object-contain"
        data-nimg="1"
        decoding="async"
        height="100"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fexample.com%2Flogo.png&w=256&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Flogo.png&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Flogo.png&w=256&q=75 2x"
        style="color: transparent;"
        width="100"
      />
    </div>
    <span
      class="text-white text-lg font-medium"
    >
      Test Logo
    </span>
  </div>
</div>
`;
